libdevcore/CommonData.h

```diff
@@ -191,7 +191,7 @@ inline unsigned bytesRequired(T _i)
 template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U const& _b)
 {
 	for (auto const& i: _b)
-		_a.push_back(i);
+		_a.emplace_back(i);
```

Comment:
This here is an example that may clarify, why the more powerful ``emplace_back`` might actually be dangerous.
Consider this:
```
class A {
};
class B {
public:
  explicit B(A const& a) {}
};
void f() {
  std::vector<B> v;
  std::vector<A> a;
  v += a;
}
```
If we use ``push_back`` in the ``operator+=`` for ``vector``, then this will fail to compile - if we use ``emplace_back``, however, it will work and will silently convert all the ``A``'s  to ``B``'s, even though the respective constructor is marked "explicit" (which one would mainly do, because this conversion is dangerous in some way).

And even in the intended valid case of:
```
std::vector<A> a;
std::vector<A> b;
a += b;
```
All elements of ``b`` will be copied anyways, even if we use ``emplace_back``, so there is no benefit in using ``emplace_back`` performance-wise.